### PR TITLE
Spiderlings will also rest when widow does

### PR DESCRIPTION
**Pull request by Zar000**
## About The Pull Request

Makes spiderlings rest when widow does
## Why It's Good For The Game

Sometimes you dont wanna kill your darlings
## Changelog
:cl:
qol: spiderlings will also rest when widow does
/:cl:

### CODE REVIEW

**Comment by iforgotmeword** [comment]
Add a option for the spideings to stay active while the vidow rests to act as meat shields

File: code/modules/mob/living/carbon/xenomorph/castes/spiderling/spiderling.dm [hunk 1]
```diff
@@ -30,6 +30,8 @@
 /datum/ai_behavior/spiderling
 	target_distance = 1
 	base_action = ESCORTING_ATOM
+	// is the spiderling currently resting?
+	var/spiderling_is_resting
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/spiderling/spiderling.dm:34]: HAS_TRAIT_FROM_ONLY(src, TRAIT_FLOORED, RESTING_TRAIT)
?

File: code/modules/mob/living/carbon/xenomorph/castes/spiderling/castedatum_spiderling.dm [hunk 2]
```diff
@@ -34,4 +34,5 @@
 
 	actions = list(
 		/datum/action/xeno_action/burrow,
+		/datum/action/xeno_action/xeno_resting,
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/spiderling/castedatum_spiderling.dm:37]: Dont you call set resting directly

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/castes/spiderling/spiderling.dm (hunk 1 above)
Comment (Zar000) [reply code/modules/mob/living/carbon/xenomorph/castes/spiderling/spiderling.dm]: I didnt use HAS_TRAIT_FROM_ONLY but maybe its ok?

**Comment by XSlayer300** [comment]
> Add a option for the spideings to stay active while the vidow rests to act as meat shields

Make it as a seperate option when right clicking the rest button, or make a keybind that has "Rest + action" into it.

File: code/modules/mob/living/carbon/xenomorph/castes/spiderling/spiderling.dm (hunk 1 above)
Comment (TiviPlus) [reply code/modules/mob/living/carbon/xenomorph/castes/spiderling/spiderling.dm]: Has a higher chance of breaking but whatever

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**